### PR TITLE
#137 RqMethod must be an interface

### PR DESCRIPTION
**Pull request by dmzaytsev**
For task #137 

* refactored RqMethod as interface
* created RqMethod.Base which implements RqMethod

### CODE REVIEW

**Comment by dmzaytsev** [comment]
@davvd Could you please find a reviewer for this pull request ?

**Comment by davvd** [comment]
@dmzaytsev Let me find a reviewer for this pull request, thanks for submitting it

**Comment by davvd** [comment]
@longtimeago it's yours, please review

**Comment by longtimeago** [comment]
@rultor merge please
@dmzaytsev perfect, thanks!

**Comment by dmzaytsev** [comment]
@longtimeago thanks for review!

File: src/main/java/org/takes/rq/RqMethod.java [hunk 1]
```diff
@@ -29,69 +29,83 @@
...
+    final class Base extends RqWrap implements RqMethod {
 
-    /**
-     * PUT method.
-     */
-    public static final String PUT = "PUT";
+        /**
+         * GET method.
+         */
+        public static final String GET = "GET";
```
Comment (yegor256) [inline src/main/java/org/takes/rq/RqMethod.java]: I think these constants should stay in `RqMethods` interface

**Comment by yegor256** [comment]
@dmzaytsev one comment above

**Comment by dmzaytsev** [comment]
@yegor256 take a look the changes please

File: src/test/java/org/takes/rq/RqMethodTest.java [hunk 2]
```diff
@@ -43,8 +43,8 @@
     @Test
     public void returnsMethod() throws IOException {
         MatcherAssert.assertThat(
-            new RqMethod(new RqFake(RqMethod.POST)).method(),
-            Matchers.equalTo(RqMethod.POST)
+            new RqMethod.Base(new RqFake(RqMethod.Base.POST)).method(),
```
Comment (yegor256) [inline src/test/java/org/takes/rq/RqMethodTest.java]: I think it should be `RqMethod.POST` here

File: src/test/java/org/takes/rq/RqMethodTest.java [hunk 3]
```diff
@@ -43,8 +43,8 @@
...
+            Matchers.equalTo(RqMethod.Base.POST)
```
Comment (yegor256) [inline src/test/java/org/takes/rq/RqMethodTest.java]: same here

**Comment by yegor256** [comment]
@dmzaytsev a few more comments above

**Comment by dmzaytsev** [comment]
@yegor256 sure
I fixed this constant

**Comment by yegor256** [comment]
@rultor merge

**Comment by davvd** [comment]
@elenavolokhova please, review this ticket for compliance with our [QA rules](http://www.teamed.io/qa.html)

**Comment by davvd** [comment]
@rultor deploy

**Comment by elenavolokhova** [comment]
@longtimeago 

According to our [quality rules](http://www.teamed.io/qa.html):
 > The code reviewer found at least three problems in the code.
 
There were no issues found during review. Try to find at least three problems next time.

Please confirm that everything is clear and you have no questions regarding this approach.

**Comment by longtimeago** [comment]
@elenavolokhova everything is clear

**Comment by elenavolokhova** [comment]
@davvd Quality is acceptable. Thanks.

**Comment by davvd** [comment]
> @davvd Quality is acceptable. Thanks.

@elenavolokhova thank you, let's try to make it "good" next time

**Comment by davvd** [comment]
@longtimeago 10 mins was added to the account of @elenavolokhova (for QA review), in transaction 56002387; **17 mins** added to your account (payment number `56002418`), many thanks for your contribution! 6 hours and 2 mins spent here.; review comments (c=2) added as a bonus; +17 added to your rating, at the moment it is: [+4688](http://www.netbout.com/b/35024?open=rating)

**Pull request merged**